EncryptionTests/Encryption Tests/CDTQIndexManagerEncryptionTests.m

```diff
@@ -0,0 +1,213 @@
+            withEncryptionKeyProvider:provider
+                                error:nil];
+
+    __unused CDTQIndexManager *im = [[CDTQIndexManager alloc] initUsingDatastore:datastore error:nil];
+
+    // Check
+    NSString *path = [CloudantSyncTests pathForQueryIndexInDatastore:datastore];
+
+    XCTAssertEqual([FMDatabase isDatabaseUnencryptedAtPath:path],
+                   kFMDatabaseUnencryptedIsUnencrypted,
+                   @"If no key is provided, index should not be encrypted");
+}
+
+- (void)testCreateQueryIndexManagerWithFixedKeyProvider
+{
+    CDTHelperFixedKeyProvider *provider = [[CDTHelperFixedKeyProvider alloc] init];
+    CDTDatastore *datastore = [self.factory datastoreNamed:@"create_query_index_tests_fixedprovider"
+                                 withEncryptionKeyProvider:provider
+                                                     error:nil];
+
+    NSError *err = nil;
+    CDTQIndexManager *im = [[CDTQIndexManager alloc] initUsingDatastore:datastore error:&err];
+
+    XCTAssertNotNil(im, @"indexManager is not nil");
+    XCTAssertNil(err, @"error has to be nil");
+}
+
+- (void)testCreateQueryIndexManagerWithFixedKeyProviderCiphersIndex
+{
+    // Create index
+    CDTHelperFixedKeyProvider *provider = [[CDTHelperFixedKeyProvider alloc] init];
+    CDTDatastore *datastore =
+        [self.factory datastoreNamed:@"create_query_index_textests_fixedprovider_cipher"
+            withEncryptionKeyProvider:provider
+                                error:nil];
+
+    __unused CDTQIndexManager *im = [[CDTQIndexManager alloc] initUsingDatastore:datastore error:nil];
+
+    // Check
+    NSString *path = [CloudantSyncTests pathForQueryIndexInDatastore:datastore];
+
+    XCTAssertEqual([FMDatabase isDatabaseUnencryptedAtPath:path], kFMDatabaseUnencryptedIsEncrypted,
+                   @"If a key is provided, index has to be encrypted");
+}
+
+- (void)testCreateQueryIndexManagerWithEncryptionKeyNilProviderFailsIfDBExistsAndItIsEncrypted
+{
+    // Create datastore
+    CDTEncryptionKeyNilProvider *provider = [CDTEncryptionKeyNilProvider provider];
+    CDTDatastore *datastore =
+        [self.factory datastoreNamed:@"create_query_index_tests_nilprovider_fails_with_cipher_db"
+            withEncryptionKeyProvider:provider
+                                error:nil];
+
+    // Create index
+    CDTQIndexManager *im = [[CDTQIndexManager alloc] initUsingDatastore:datastore error:nil];
+
+    // Replace index
+    NSFileManager *defaultManager = [NSFileManager defaultManager];
+    NSString *indexPath = [CloudantTests pathForQueryIndexInDatastore:datastore];
+
+    [defaultManager removeItemAtPath:indexPath error:nil];
+
+    NSBundle *bundle = [NSBundle bundleForClass:[self class]];
+    NSString *assetPath = [bundle pathForResource:@"emptyencryptedindex" ofType:@"sqlite"];
+
+    [[NSFileManager defaultManager] copyItemAtPath:assetPath toPath:indexPath error:nil];
```

Comment:
I'm not really understanding why this test requires that we create the index database first before replacing it. Can we just create the database, then put in the canned database before initialising the IndexManager object?